backends/vulkan/runtime/api/containers/Tensor.cpp

```diff
@@ -1191,10 +1242,10 @@ void vTensor::virtual_transpose(const int64_t dim0, const int64_t dim1) {
 
   const int dim0_whcn = sizes_.size() - 1 - dim0;
   const int dim1_whcn = sizes_.size() - 1 - dim1;
-  if (packed_dim_ == dim0_whcn) {
-    packed_dim_ = dim1_whcn;
-  } else if (packed_dim_ == dim1_whcn) {
-    packed_dim_ = dim0_whcn;
+  if (packed_dim_info_.packed_dim == dim0_whcn) {
```

Comment:
virtual_transpose() function updates packed_dim when one of the transposed dimensions is the packed dimension, but it doesn't update outer_packed_dim for tiled layouts.

Either disallow transpose for tiled layouts (assert explicitly) or support it.